pyomo/contrib/parmest/ScenarioCreator.py

```diff
@@ -0,0 +1,211 @@
+                f.write(",{}".format(n))
+            f.write('\n')
+            for s in self.ScensIterator():
+                f.write("{},{}".format(s.name, s.probability))
+                for v in s.ThetaVals.values():
+                    f.write(",{}".format(v))
+                f.write('\n')
+
+
+class ParmestScen(object):
+    """ A little container for scenarios; the Args are the attributes.
+
+    Args:
+        name (str): name for reporting; might be ""
+        ThetaVals (dict): ThetaVals[name]=val
+        probability (float): probability of occurance "near" these ThetaVals
+    """
+
+    def __init__(self, name, ThetaVals, probability):
+        self.name = name
+        assert(isinstance(ThetaVals, dict))
+        self.ThetaVals = ThetaVals
+        self.probability = probability
+
+############################################################
+
+
+class ScenarioCreator(object):
+    """ Create scenarios from parmest.
+
+    Args:
+        pest (Estimator): the parmest object
+        solvername (str): name of the solver (e.g. "ipopt")
+
+    """
+
+    def __init__(self, pest, solvername):
+        self.pest = pest
+        self.solvername = solvername
+        self.experiment_numbers = pest._numbers_list
+
+
+    def ScenariosFromExperiments(self, addtoSet):
+        """Creates new self.Scenarios list using the experiments only.
+
+        Args:
+            addtoSet (ScenarioSet): the scenarios will be added to this set
+        Returns:
+            a ScenarioSet
+        """
+
+        assert(isinstance(addtoSet, ScenarioSet))
+        prob = 1. / len(self.pest._numbers_list)
+        for exp_num in self.pest._numbers_list:
+            ##print("Experiment number=", exp_num)
+            model = self.pest._instance_creation_callback(exp_num,
+                                                        self.pest.callback_data)
+            opt = pyo.SolverFactory(self.solvername)
+            results = opt.solve(model)  # solves and updates model
+            ## pyo.check_termination_optimal(results)
+            ThetaVals = dict()
+            for theta in self.pest.theta_names:
+                tvar = eval('model.'+theta)
+                tval = pyo.value(tvar)
+                ##print("    theta, tval=", tvar, tval)
+                ThetaVals[theta] = tval
+            addtoSet.addone(ParmestScen("ExpScen"+str(exp_num), ThetaVals, prob))
+            
+    def ScenariosFromBoostrap(self, addtoSet, numtomake, seed=None):
+        """Creates new self.Scenarios list using the experiments only.
+
+        Args:
+            addtoSet (ScenarioSet): the scenarios will be added to this set
+            numtomake (int) : number of scenarios to create
+        """
+
+        assert(isinstance(addtoSet, ScenarioSet))
+
+        bootstrap_thetas = self.pest.theta_est_bootstrap(numtomake, seed=seed)
+        addtoSet.append_bootstrap(bootstrap_thetas)
+
+        
+if __name__ == "__main__":
```

Comment:
I think this was moved into a unittest.  Can you delete it from here (if for no other reason to avoid code coverage issues)?